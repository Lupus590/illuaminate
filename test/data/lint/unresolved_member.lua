-- config: (lint (only var:unresolved-member))

--- This is a documented term
local tbl_1 = { x = 1 }

-- This is not documented
local tbl_2 = { x = 1 }

--- This is also documented
local function not_tbl() end

print(tbl_1.x, tbl_1.y)
print(tbl_2.x, tbl_2.y) -- Doesn't warn as not documented
print(not_tbl.x) -- Doesn't warn as not table
