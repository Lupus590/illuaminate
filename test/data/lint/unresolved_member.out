unresolved_member.lua:[12:16-12:22]: Unknown field y in tbl_1 [var:unresolved-member]
    │
 12 │ print(tbl_1.x, tbl_1.y)
    │                ^^^^^^^
unresolved_member.lua:[19:9-19:15]: Unknown field y in tbl_3 [var:unresolved-member]
    │
 19 │   print(tbl_3.y)
    │         ^^^^^^^
unresolved_member.lua:[24:9-24:15]: Unknown field y in tbl_4 [var:unresolved-member]
    │
 24 │   print(tbl_4.y)
    │         ^^^^^^^
unresolved_member.lua:[29:7-29:22]: Unknown field unknown in a_module [var:unresolved-member]
    │
 29 │ print(a_module.unknown) --- Undocumented
    │       ^^^^^^^^^^^^^^^^
No errors and 4 warnings
